diff --git a/node_modules/astro-integration-lottie/Lottie.astro b/node_modules/astro-integration-lottie/Lottie.astro
index 20a445f..f0bf75b 100644
--- a/node_modules/astro-integration-lottie/Lottie.astro
+++ b/node_modules/astro-integration-lottie/Lottie.astro
@@ -6,4 +6,4 @@ export interface Props extends LottieAnimationConfig {}
 const { ...lottieData } = Astro.props;
 ---
 
-<div data-lottie data-lottie-data={JSON.stringify(lottieData)}></div>
+<div id={lottieData.id} data-lottie data-lottie-data={JSON.stringify(lottieData)}></div>
diff --git a/node_modules/astro-integration-lottie/dist/loader.js b/node_modules/astro-integration-lottie/dist/loader.js
index b0b5f51..2d09750 100644
--- a/node_modules/astro-integration-lottie/dist/loader.js
+++ b/node_modules/astro-integration-lottie/dist/loader.js
@@ -1,18 +1,19 @@
-setTimeout(async () => {
+async function loadLottie(idToRender) {
     const DEFAULT = {
         player: "light",
         loop: true,
         autoplay: "visible",
         visibleThreshold: 0,
     };
-    const containers = [...document.querySelectorAll("[data-lottie]")].map(x => {
+    let containers = [...document.querySelectorAll("[data-lottie]")].map(x => {
         try {
             return [x, { ...DEFAULT, ...JSON.parse(x.getAttribute("data-lottie-data") || "") }];
         }
         catch (err) {
             console.warn("Cannot parse lottie animation data", x);
         }
-    }).filter(x => !!x);
+    }).filter(x => !!x)
+    
     if (containers.length === 0) {
         // no lottie animation return
         return;
@@ -47,9 +48,10 @@ setTimeout(async () => {
         const animationData = animationDataMap.get(config.src);
         let player;
         if (animationData) {
+            container.innerHTML = '';
             const { loop, autoplay } = config;
             player = lottie.loadAnimation({
-                container,
+               container,
                 loop,
                 autoplay: autoplay === "visible" ? false : autoplay,
                 animationData,
@@ -112,5 +114,8 @@ setTimeout(async () => {
     document.dispatchEvent(new CustomEvent("astro-lottie-loaded", {
         detail: window.astroLottie
     }));
-}, 0);
-export {};
+}
+window.loadLottie = loadLottie
+setTimeout(loadLottie, 0);
+export { };
+
